### DISCUSSION TITLE
Are there full examples not using FeynArts?

### DISCUSSION THREAD

**Discussion by foice** [FeynCalc/feyncalc#83] [Q&A]
Hello there! I see a huge list of examples in [Gallery](https://feyncalc.github.io/examples), that is very useful. However it seems that all the examples I have opened are based on matrix element calculations from FeynArts.

Is there any full process calculation based on amplitudes written by hand?

Of course the Reference Guide provides the bits and pieces one has to patch together to write an amplitude, but it is not impossible to get stuck in pitfalls. So I thought a full example including the writing of the amplitude using FeynCalc technology might be very useful.

**Comment by vsht** [comment]
Not really. The reason being that writing amplitudes by hand (at least within FeynCalc) 
is not so convenient. 

What kind of processes do you actually have in mind? Text-book level stuff or
something related to the real-life research?

The only thing I could imagine where one would enter something by hand
could be scattering amplitudes in theories where everything is cut constructible.
Although even there people most likely use automation for this step.

For anything else FeynArts and Qgraf can actually cater for all your needs.

**Reply by foice** [reply]
Thanks for replying. I do have in mind both text-book tree-level stuff and more real-life things. In both cases showing how to write and manipulate amplitudes has pedagogical benefits and it may be useful for research as well. 

- About the textbook processes, e.g. `$e^+e- \to \gamma \gamma$`. In my opinion it makes a lot of sense these days to parallel blackboard QFT and phenomenology classes with some showcasing of modern tools like FeynCalc. This will allow a much gentler intro into research (when it comes the time) and will allow new ways to explore the subject for the more proactive and curious students.

- For processes in which new physics is involved, going all the way to write a FeynArts model may not be worth.  I am thinking tree-level, as it is most times what you need for new physics. Imagine I just need to replace a photon with another (possibly massive) U(1) gauge boson and I am willing to take the risk of writing the amplitude myself. That would be easier if I can see example calculations not using FeynArts output to teach me a bit how to deal with the algebra and write this simple amplitudes in minutes, with all signs and factors 2 in the right place ;) 

- Somewhere between real-life research and teaching (oneself), I may want to see how exactly my (BSM) amplitude behaves in a given context. To my knowledge this can be done in FeynArts using `VecSet`  and `ToComponents` (if I remember correctly the names), but it still requires the FeynArts model. I have instead enjoyed writing down the amplitude in FeynCalc and then plug the necessary spinors and polarization vectors to play with the amplitude "as a complex number". A little help to avoid common pitfalls here would make things more accessible to many.

So I think that FeynCalc offers some nice way to deal with simple calculations even starting from scratch by writing the amplitude yourself and then get algebra done in an automated way. A demo of this way of operation would benefit many, that's my opinion.

**Reply by vsht** [reply]
> About the textbook processes, e.g. $e^+e- \to \gamma \gamma$. In my opinion it makes a lot of sense these days to parallel blackboard QFT and phenomenology classes with some showcasing of modern tools like FeynCalc. This will allow a much gentler intro into research (when it comes the time) and will allow new ways to explore the subject for the more proactive and curious students.


Most likely https://porthos.tecnico.ulisboa.pt/CTQFT/node2.html
provides the kind of examples you may be looking for.

I don't think that there are much pitfalls at the tree-level, as long as 
you pay attention to using proper noncommutative multiplication
when needed (`Dot` vs `Times`) and avoiding indices that violate
Einstein's summation. That is, provided that the amplitude you enter
is fully correct per se.

> For processes in which new physics is involved, going all the way to write a FeynArts model may not be worth. 

Once you learn how to use FeynRules, you would be able to implement a new FeynArts 
model in no time. FeynRules is also what the majority of BSM folks use to explore their
models (usually in conjecture with MadGraph, not FeynCalc)

> That would be easier if I can see example calculations not using FeynArts output to teach me a bit how to deal with the algebra and write this simple amplitudes in minutes, with all signs and factors 2 in the right place ;)

Getting all the signs and factors right is exactly what a diagram
generator gives you for free. Otherwise you essentially have the following possibilities

1. Derive the Feynman rules by hand and enter the amplitudes according to those
rules (again by hand).
2. Define some custom functions encoding the Feynman rules and enter the amplitudes
as products of those functions.

In both cases you have to hope that you didn't make any obvious or nonobvious mistakes.
I'm not aware of a magic trick which ensures that the given amplitude in any theory 
entered by hand is 100% correct.

> A demo of this way of operation would benefit many, that's my opinion.

I'm afraid I'm not the right person to develop this kind of examples.  I always stick to diagram
generators unless it is for manipulating some standalone expressions.

**Reply by foice** [reply]
Thanks for the reference, seems to go in the direction I meant! 

Just to mention a pitfall I met in my recent attempts:
```
QuarkPropagator[ke - kV, Explicit -> True, Dimension -> 4]
```
If the Dimension option is not given you get an error. This is kind of easy to spot with `FCGetDimensions[]`, once you realize the source of the issue (as with most debugging). There might be others I have not met yet of course. Your contact with beginners may contribute to the list.

FeynRules is great, I am aware of it, but it has its own quirks and overhead. In any case, at least in my practice, is not uncommon that going to FeynRules is just not worth for the same reason I said it is not worth involving FeynArts. You are there and you want to see what the amplitude is going to be in a bunch of different cases for which it is just not worth building a whole FA or FR or UFO model for each of them. Having some guidance on how to use the Spinor, the Gamma matrices, the propagators and so on can give people a kick-start. I think we are talking about few basic QED and EW processes that's all.

Of course FeynArts and other diagram generators cannot be replaced for a final computation, but I guess you see that I am talking about the level of complexity around  that of textbook calculations, be the target a textbook calculation or something BSM as simple as that. Plus ... even a single diagram process can benefit of this!

**Reply by vsht** [reply]
Hmm, perhaps I'm somehow missing your point, but it seems to me that the aspects
you mention belong to the basic familiarity with FeynCalc. 

It's true that `QuarkPropagator` and alike are d-dim by default. But normally one would immediately
notice that already from the typesetting (no bars on top of vertices and Dirac matrices).

Then you write that
> Having some guidance on how to use the Spinor, the Gamma matrices, the propagators and so on can give people a kick-start.

But how is it related to the way the amplitude was obtained? I mean, if one is not comfortable
manipulating those objects, it won't really help to have the amplitude from FeynArts. 

To me it seems that if you go through the tutorial (FeynCalc/Tutorials/Intro in the Documentation center)
and the available examples, you should already be familiar with most of the stuff.
 
Though the usage of `QuarkPropagator`, `QuarkGluonVertex` etc. is not covered there, I'm not sure that
those functions are really the most convenient way to enter amplitudes by hand. As I wrote, I don't usually
do that myself, so perhaps I'm lacking a good insight into bootstrapping such calculations.

**Reply by foice** [reply]
Exactly, basic familiarity about FeynCalc is largely demonstrated on what happens once the amplitude is somehow given to you from FeynArts. Instead I might decide to write it on my own or process it in ways that are following the template you also present in the documentation.

My limited experience is to play with things like
```
M = Contract[
  SpinorVBar[Momentum[kE], 0].GA[\[Mu]].QuarkPropagator[ke - kV, 
    Explicit -> True, Dimension -> 4].GA[\[Nu]].GA[7].SpinorU[
    Momentum[ke], 0]]
Mstar = ComplexConjugate[
   M] //. {LorentzIndex[\[Mu]] -> LorentzIndex[mu], 
   LorentzIndex[\[Nu]] -> LorentzIndex[nu]}
FermionSpinSum[
  Mstar M* PolarizationSum[\[Mu], mu, kW]*
   PolarizationSum[\[Nu], nu, kV]];
DiracSimplify[%]
```
which are coming from earlier versions Help and stratification on it. Since now there is a very comprehensive list of processes, all falling under the [same template](https://github.com/FeynCalc/feyncalc/tree/hotfix-stable/FeynCalc/Examples#example-file-structure), what I was suggesting is to add a few more examples that go along another template, e.g. they amplitude can be computed from its FeynCalc building blocks. There are probably other interesting variations that can be thought of. I think having fully fleshed examples will help beginners and students who need or want to follow a different path than the current template.

**Reply by vsht** [reply]
Well, I suppose there are all kind of examples one could add, but as you can see we are quite short
on manpower and there are still a lot of things to add/improve that are more urgent.

You are of course free to post your examples here, then I could link to this thread when similar requests arise.

**Reply by vsht** [reply]
There are actually some interesting nuclear physics examples that I would like to
work out when I find time (#84). Those would probably fit into your request for codes not
using FeynArts.

As far as the tree-level QED is concerned, the lecture notes of Jorge Romao are 
good enough and I see no point to duplicate that.
